spaceship.zsh

```diff
@@ -405,39 +405,89 @@ spaceship_dir() {
 }
 
 # GIT BRANCH
-# Show current git brunch using git_current_status from Oh-My-Zsh
+# Show current git branch
 spaceship_git_branch() {
   [[ $SPACEHIP_GIT_BRANCH_SHOW == false ]] && return
 
   _is_git || return
 
+  local ref
+  ref=$(command git symbolic-ref --quiet HEAD 2> /dev/null)
+  local ret=$?
+  if [[ $ret != 0 ]]; then
+    [[ $ret == 128 ]] && return
+    ref=$(command git rev-parse --short HEAD 2> /dev/null) || return
+  fi
+
+  local git_current_branch=${ref#refs/heads/}
+
   _prompt_section \
     "$SPACESHIP_GIT_BRANCH_COLOR" \
     "$SPACESHIP_GIT_BRANCH_PREFIX$(git_current_branch)$SPACESHIP_GIT_BRANCH_SUFFIX"
 }
 
 # GIT STATUS
-# Check if current dir is a git repo, set up ZSH_THEME_* variables
-# and show git status using git_prompt_status from Oh-My-Zsh
-# Reference:
-#   https://github.com/robbyrussell/oh-my-zsh/blob/master/lib/git.zsh
+# Show git status
 spaceship_git_status() {
   [[ $SPACESHIP_GIT_STATUS_SHOW == false ]] && return
 
   _is_git || return
 
-  ZSH_THEME_GIT_PROMPT_UNTRACKED=$SPACESHIP_GIT_STATUS_UNTRACKED
-  ZSH_THEME_GIT_PROMPT_ADDED=$SPACESHIP_GIT_STATUS_ADDED
-  ZSH_THEME_GIT_PROMPT_MODIFIED=$SPACESHIP_GIT_STATUS_MODIFIED
-  ZSH_THEME_GIT_PROMPT_RENAMED=$SPACESHIP_GIT_STATUS_RENAMED
-  ZSH_THEME_GIT_PROMPT_DELETED=$SPACESHIP_GIT_STATUS_DELETED
-  ZSH_THEME_GIT_PROMPT_STASHED=$SPACESHIP_GIT_STATUS_STASHED
-  ZSH_THEME_GIT_PROMPT_UNMERGED=$SPACESHIP_GIT_STATUS_UNMERGED
-  ZSH_THEME_GIT_PROMPT_AHEAD=$SPACESHIP_GIT_STATUS_AHEAD
-  ZSH_THEME_GIT_PROMPT_BEHIND=$SPACESHIP_GIT_STATUS_BEHIND
-  ZSH_THEME_GIT_PROMPT_DIVERGED=$SPACESHIP_GIT_STATUS_DIVERGED
-
-  local git_status="$(git_prompt_status)"
+  local INDEX STATUS
+  INDEX=$(command git status --porcelain -b 2> /dev/null)
+  git_status=""
+
+  if $(echo "$INDEX" | command grep -E '^\?\? ' &> /dev/null); then
```

Comment:
Let's add some comments before each of these conditions explaining what are they for.